torch/csrc/distributed/c10d/CUDASymmetricMemory.cu

```diff
@@ -1,275 +1,32 @@
-    auto cmsg = CMSG_FIRSTHDR(&msg);
-    cmsg->cmsg_len = CMSG_LEN(sizeof(int));
-    cmsg->cmsg_level = SOL_SOCKET;
-    cmsg->cmsg_type = SCM_RIGHTS;
-
-    if (fd != -1) {
-      std::copy(
-          reinterpret_cast<const char*>(&fd),
-          reinterpret_cast<const char*>(&fd) + sizeof(fd),
-          reinterpret_cast<char*>(CMSG_DATA(cmsg)));
-    } else {
-      msg.msg_controllen = 0;
-    }
-
-    TORCH_CHECK(
-        sendmsg(socket_, &msg, 0) > 0,
-        "Failed to send fd: ",
-        c10::utils::str_error(errno));
-  }
-
-  int recv_fd() {
-    char buf[2];
-    struct iovec io = {.iov_base = (void*)buf, .iov_len = sizeof(buf)};
-
-    char cbuf[CMSG_SPACE(sizeof(int))];
-    memset(cbuf, 0, sizeof(cbuf));
-
-    struct msghdr msg = {
-        .msg_iov = &io,
-        .msg_iovlen = 1,
-        .msg_control = cbuf,
-        .msg_controllen = sizeof(cbuf)};
-
-    TORCH_CHECK(
-        recvmsg(socket_, &msg, 0) > 0,
-        "Failed to receive fd: ",
-        c10::utils::str_error(errno));
-
-    if (msg.msg_controllen == 0) {
-      return -1;
-    }
-
-    auto cmsg = CMSG_FIRSTHDR(&msg);
-    TORCH_CHECK(cmsg != NULL);
-    TORCH_CHECK(cmsg->cmsg_len == CMSG_LEN(sizeof(int)));
-    TORCH_CHECK(
-        cmsg->cmsg_level == SOL_SOCKET && cmsg->cmsg_type == SCM_RIGHTS);
-    return *reinterpret_cast<int*>(CMSG_DATA(cmsg));
-  }
-
-  std::vector<int> all_gather_fds(
-      int rank,
-      const std::vector<int>& pids,
-      int fd) {
-    size_t world_size = pids.size();
-    std::vector<int> fds(pids.size());
-    fds[rank] = fd;
-
-    int dst_rank = (rank + 1) % world_size;
-    for (size_t step = 1; step < world_size; ++step) {
-      int src_rank = (rank + world_size - step) % world_size;
-      send_fd(pids[dst_rank], fd);
-      fd = recv_fd();
-      fds[src_rank] = fd;
-    }
-    return fds;
-  }
-
-  int broadcast_fds(
-      int rank,
-      int src_rank,
-      const std::vector<int>& pids,
-      int fd) {
-    size_t world_size = pids.size();
-
-    if (rank == src_rank) {
-      for (int dst_rank = 0; dst_rank < (int)world_size; ++dst_rank) {
-        if (dst_rank == rank) {
-          continue;
-        }
-        send_fd(pids[dst_rank], fd);
-      }
-      return fd;
-    }
-    return recv_fd();
-  }
-
- private:
-  static std::string get_socket_name(int pid) {
-    std::string tmp_dir = "/tmp";
-    for (const char* env_var : {"TMPDIR", "TMP", "TEMP", "TEMPDIR"}) {
-      if (const auto path = c10::utils::get_env(env_var)) {
-        tmp_dir = path.value();
-        break;
-      }
-    }
-    std::ostringstream oss;
-    oss << tmp_dir << "/symm_mem-" << pid;
-    return oss.str();
-  }
+namespace c10d {
+namespace symmetric_memory {
 
-  std::string socket_name_;
-  int socket_;
-};
+/* Start of CUDASymmetricMemory implementation */
 
-constexpr size_t signal_pad_size = 2048;
 const std::string store_comm_prefix = "CUDASymmetricMemory";
 
-static size_t store_comm_seq_id = 0;
-
-template <typename T>
```

Comment:
3 functions moved to utils file; not checking the details, assuming its just a direct move